### PR TITLE
[docs] add InfiniteOpt to documentation

### PR DESCRIPTION
**Pull request by odow**
https://github.com/jump-dev/JuMP.jl/pull/3342 but on the `jump-dev` remote so that we can get a doc preview. I'll post the link once CI finishes.

https://jump.dev/JuMP.jl/previews/PR3343/packages/InfiniteOpt/

### CODE REVIEW

File: docs/packages.toml [hunk 1]
```diff
@@ -120,6 +120,11 @@
 #   user = "GAMS-dev"
 # [Hypatia]
 #   user = "chriscoey"
+[InfiniteOpt]
+  user = "infiniteopt"
+  rev = "c280bf81cb8144077faec11d16bad7b5a09b0f6f"
```
Comment (odow) [inline docs/packages.toml:125]:
```suggestion
  rev = "jump-readme"
```

**Comment by odow** [comment]
@pulsipher: now the problem. The iteration time to tweak little parts of the docs is really long :cry:

**Comment by pulsipher** [comment]
> @pulsipher: now the problem. The iteration time to tweak little parts of the docs is really long 😢

Yes the ~40min runtime for the docs is pretty painful, but on the bright side the docs have a lot of good stuff now :)

**Comment by odow** [comment]
Docs are updated and all checks are green. We can either wait for a tag, or if you merge we can just use a commit revision for now.

File: docs/packages.toml [hunk 2]
```diff
@@ -120,6 +120,11 @@
 #   user = "GAMS-dev"
 # [Hypatia]
 #   user = "chriscoey"
+[InfiniteOpt]
+  user = "infiniteopt"
+  rev = "jump-readme"
```
Comment (odow) [inline docs/packages.toml:125]: Change before merging

Comment (pulsipher) [reply docs/packages.toml:125]: I'll change it to the new tag which will be `v0.5.7`. I should have the new version tagged within an hour or so, I want to release some patch fixes anyway.

Comment (odow) [reply docs/packages.toml:125]: Cool. Ping here when you're ready for a rebuild.

Comment (pulsipher) [reply docs/packages.toml:125]: Ok, `v0.5.7` is released.

**Pull request merged**